homeassistant/components/light/knx.py

```diff
@@ -154,19 +233,57 @@ def supported_features(self):
         if self.device.supports_brightness:
             flags |= SUPPORT_BRIGHTNESS
         if self.device.supports_color:
-            flags |= SUPPORT_COLOR
+            flags |= SUPPORT_COLOR | SUPPORT_BRIGHTNESS
+        if self.device.supports_color_temperature or \
+                self.device.supports_tunable_white:
+            flags |= SUPPORT_COLOR_TEMP
         return flags
 
     async def async_turn_on(self, **kwargs):
         """Turn the light on."""
-        if ATTR_BRIGHTNESS in kwargs:
-            if self.device.supports_brightness:
-                await self.device.set_brightness(int(kwargs[ATTR_BRIGHTNESS]))
-        elif ATTR_HS_COLOR in kwargs:
-            if self.device.supports_color:
-                await self.device.set_color(color_util.color_hs_to_RGB(
-                    *kwargs[ATTR_HS_COLOR]))
+        brightness = kwargs.get(ATTR_BRIGHTNESS, self.brightness)
+        hs_color = kwargs.get(ATTR_HS_COLOR, self.hs_color)
+        mireds = kwargs.get(ATTR_COLOR_TEMP, self.color_temp)
+
+        # fall back to default values, if required
```

Comment:
I'm sure @oliverblaha knows why.

Reply:
I think it is required when for RGB lights. Eg. when hs_color shall be changed but brightness is not yet known. Otherwise `color_util.color_hsv_to_RGB(*hs_color, brightness * 100 / 255)` in line 269 would fail. 
Can somebody confirm this - @Julius2342 maybe?

Reply:
Sorry, missed that conversation before!
Yes, that's exactly when you need it. For converting to RGB, you always need hue, saturation and brightness; however, ha might send hue and saturation only. If that is the case, then the current brightness should be used. If that is not known, there's no other choice than taking some default value (or, well, just do nothing).

Reply:
To be clear, I am not saying that the code is wrong. Just that "if required" carries no information and should be replaced with text actually saying when it is required.

Reply:
@oliverblaha thanks for your reply! So I guess we can move the fall back tests inside of 265
```python
elif self.device.supports_color and \
                (update_brightness or update_color):
            # change RGB color (includes brightness)
```
to make this clearer?

Reply:
I would recommend against moving it in there. Technically it would (currently) be the best position, however, I believe it would be much harder to read - especially as the brightness might be set from three different sources (new value, existing value or default value).

That's why I chose to make sure that every variable is properly defined at the beginning of the method. There might be other scenarios later (e.g. when adding support for KNX devices that need HSV to set their color) which also need a default brightness if not set otherwise.

So I'd suggest to leave it where it is and change the comment as follows:
# to make sure that the following calculations always have valid values available, fall back to default values for any parameter that has neither been set to a new value nor has an known current value.

Reply:
Even if HSV devices were supported, HA would still address color changes the same way (through our update_color attribute). I would suggest to move it for now. We can always revert it if there is another usecase of this defaults.